projects/packages/licensing/src/class-licensing.php

```diff
@@ -269,4 +273,381 @@ public function get_license_activation_notice_dismiss() {
+			)
+		);
+
+		/**
+		 * Get Jetpack user license counts.
+		 */
+		register_rest_route(
+			'jetpack/v4',
+			'licensing/user/counts',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_user_license_counts',
+				'permission_callback' => __CLASS__ . '::user_licensing_permission_check',
+			)
+		);
+
+		/**
+		 * Update user-licensing activation notice dismiss info.
+		 */
+		register_rest_route(
+			'jetpack/v4',
+			'licensing/user/activation-notice-dismiss',
+			array(
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::update_licensing_activation_notice_dismiss',
+				'permission_callback' => __CLASS__ . '::user_licensing_permission_check',
+				'args'                => array(
+					'last_detached_count' => array(
+						'required'          => true,
+						'type'              => 'integer',
+						'validate_callback' => __CLASS__ . '::validate_non_neg_int',
+					),
+				),
+			)
+		);
+
+		/**
+		 * Attach licenses to user account
+		 */
+		register_rest_route(
+			'jetpack/v4',
+			'/licensing/attach-licenses',
+			array(
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::attach_jetpack_licenses',
+				'permission_callback' => __CLASS__ . '::user_licensing_permission_check',
+				'args'                => array(
+					'licenses' => array(
+						'required' => true,
+						'type'     => 'array',
+						'items'    => array(
+							'type' => 'string',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Verify that the user can set a Jetpack license key
+	 *
+	 * @since 9.5.0
+	 *
+	 * @return bool|WP_Error True if user is able to set a Jetpack license key
+	 */
+	public static function set_jetpack_license_key_permission_check() {
+		if ( self::instance()->is_licensing_input_enabled() ) {
+			return true;
+		}
+
+		return new WP_Error( 'invalid_user_permission_set_jetpack_license_key', self::$user_permissions_error_msg, array( 'status' => rest_authorization_required_code() ) );
+
+	}
+
+	/**
+	 * Verify that user can view and update user-licensing data.
+	 *
+	 * @return bool Whether the user is currently connected and they are the connection owner.
+	 */
+	public static function user_licensing_permission_check() {
+		$connection_manager = new Connection_Manager( 'jetpack' );
+
+		if ( $connection_manager->is_user_connected() && $connection_manager->is_connection_owner() ) {
+			return true;
+		}
+
+		return new WP_Error( 'invalid_permission_manage_user_licenses', REST_Connector::get_user_permissions_error_msg(), array( 'status' => rest_authorization_required_code() ) );
+	}
+
+	/**
+	 * Verify that user can view Jetpack admin page.
+	 *
+	 * @since 4.3.0
+	 *
+	 * @return bool Whether user has the capability 'jetpack_admin_page'.
+	 */
+	public static function view_admin_page_permission_check() {
+		if ( current_user_can( 'jetpack_admin_page' ) ) {
```

Comment:
Since this may be in a different plugin, maybe we should use more standard permissions, such as `manage_options`?